From 969bafa9047b37a3eab5de4d78ff9496c995f4d1 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 29 Aug 2025 08:50:31 +0300
Subject: [PATCH 072/N] configure: define _DEBUG for a debug build

The Windows code expects it in some places for a debug build

Skip some things which fail to link with _DEBUG
---
 Modules/_winapi.c          | 2 +-
 PC/clinic/msvcrtmodule.c.h | 4 ++--
 configure.ac               | 1 +
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Modules/_winapi.c b/Modules/_winapi.c
index cbd85ea..65c9cfb 100644
--- a/Modules/_winapi.c
+++ b/Modules/_winapi.c
@@ -1494,7 +1494,7 @@ static PyObject *
 _winapi_ExitProcess_impl(PyObject *module, UINT ExitCode)
 /*[clinic end generated code: output=a387deb651175301 input=4f05466a9406c558]*/
 {
-    #if defined(Py_DEBUG)
+    #if defined(Py_DEBUG) && defined(_MSC_VER)
 #ifdef MS_WINDOWS_DESKTOP
         SetErrorMode(SEM_FAILCRITICALERRORS|SEM_NOALIGNMENTFAULTEXCEPT|
                      SEM_NOGPFAULTERRORBOX|SEM_NOOPENFILEERRORBOX);
diff --git a/PC/clinic/msvcrtmodule.c.h b/PC/clinic/msvcrtmodule.c.h
index a77d085..4a64c5b 100644
--- a/PC/clinic/msvcrtmodule.c.h
+++ b/PC/clinic/msvcrtmodule.c.h
@@ -517,7 +517,7 @@ exit:
 
 #endif /* defined(MS_WINDOWS_DESKTOP) */
 
-#if defined(_DEBUG)
+#if defined(_DEBUG) && defined(_MSC_VER)
 
 PyDoc_STRVAR(msvcrt_CrtSetReportFile__doc__,
 "CrtSetReportFile($module, type, file, /)\n"
@@ -564,7 +564,7 @@ exit:
 
 #endif /* defined(_DEBUG) */
 
-#if defined(_DEBUG)
+#if defined(_DEBUG) && defined(_MSC_VER)
 
 PyDoc_STRVAR(msvcrt_CrtSetReportMode__doc__,
 "CrtSetReportMode($module, type, mode, /)\n"
diff --git a/configure.ac b/configure.ac
index 41f2e70..01e1422 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2389,6 +2389,7 @@ _RESTORE_VAR([CFLAGS])
 PYDEBUG_CFLAGS="-O0"
 AS_VAR_IF([ac_cv_cc_supports_og], [yes],
           [PYDEBUG_CFLAGS="-Og"])
+PYDEBUG_CFLAGS="-D_DEBUG $PYDEBUG_CFLAGS"
 
 # gh-120688: WASI uses -O3 in debug mode to support more recursive calls
 if test "$ac_sys_system" = "WASI"; then
